astropy/io/fits/tests/test_fitsdiff.py

```diff
@@ -57,9 +45,73 @@ def test_onediff(self):
         tmp_b = self.temp('testb.fits')
         hdu_a.writeto(tmp_a)
         hdu_b.writeto(tmp_b)
-        testargs = [tmp_a, tmp_b]
-        sys.argv += testargs
-        numdiff = fitsdiff.main()
+        numdiff = fitsdiff.main([tmp_a, tmp_b])
+        assert numdiff == 1
+
+    def test_manydiff(self, capsys):
+        a = np.arange(100).reshape((10, 10))
+        hdu_a = PrimaryHDU(data=a)
+        b = a + 1
+        hdu_b = PrimaryHDU(data=b)
+        tmp_a = self.temp('testa.fits')
+        tmp_b = self.temp('testb.fits')
+        hdu_a.writeto(tmp_a)
+        hdu_b.writeto(tmp_b)
+
+        numdiff = fitsdiff.main([tmp_a, tmp_b])
+        out, err = capsys.readouterr()
+        assert numdiff == 1
+        assert out.splitlines()[-4:] == [
+            '        a> 9',
+            '        b> 10',
+            '     ...',
+            '     100 different pixels found (100.00% different).']
+
+        numdiff = fitsdiff.main(['-n', '1', tmp_a, tmp_b])
+        out, err = capsys.readouterr()
+        assert numdiff == 1
+        assert out.splitlines()[-4:] == [
+            '        a> 0',
+            '        b> 1',
+            '     ...',
+            '     100 different pixels found (100.00% different).']
+
+    def test_outputfile(self):
+        a = np.arange(100).reshape((10, 10))
+        hdu_a = PrimaryHDU(data=a)
+        b = a.copy()
+        b[1, 0] = 12
+        hdu_b = PrimaryHDU(data=b)
+        tmp_a = self.temp('testa.fits')
+        tmp_b = self.temp('testb.fits')
+        hdu_a.writeto(tmp_a)
+        hdu_b.writeto(tmp_b)
+
+        numdiff = fitsdiff.main(['-o', self.temp('diff.txt'), tmp_a, tmp_b])
+        assert numdiff == 1
+        with open(self.temp('diff.txt')) as f:
+            out = f.read()
+        assert out.splitlines()[-4:] == [
+            '     Data differs at [1, 2]:',
+            '        a> 10',
+            '        b> 12',
+            '     1 different pixels found (1.00% different).']
+
+    def test_atol(self):
+        a = np.arange(100, dtype=np.float).reshape((10, 10))
```

Comment:
You don't need the `()` inside the reshape `reshape(10, 10)` should also work

Reply:
Sure. While you were here you could also note the `np.float` :stuck_out_tongue:

Reply:
oh, did I forget to submit that comment? 😓